src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -447,7 +461,16 @@ func (r *StatefulSetResource) obj(
 									Name:      "config-dir",
 									MountPath: configDestinationDir,
 								},
+								{
+									Name:      "hooks-dir",
+									MountPath: hooksMountDir,
+								},
 							}, tlsVolumeMounts...),
+							Lifecycle: &corev1.Lifecycle{
+								PostStart: r.getPostStartHook(),
+								PreStop:   r.getPreStopHook(),
+							},
+							ReadinessProbe: r.getReadinessProbe(),
```

Comment:
It would be nice if commit message include the reason why readiness probe is introduced. The initial design was assuming that unready brokers should be exposed regardless of the internal state. Kafka clients would retries if necessary. If this is added I would assume we need to change the Service definition
https://github.com/redpanda-data/redpanda/blob/8b4d589bc7ea8b52c9bffdc9268a0a6aad9cfe5b/src/go/k8s/pkg/resources/headless_service.go#L110